youtube_downloader/proxy_manager.py

```
@@ -0,0 +1,326 @@
+            return None
+        
+        selected = random.choice(healthy_proxies)
+        selected.last_used = time.time()
+        return selected
+    
+    def record_success(self, proxy: ProxyConfig):
+        """Record a successful request using this proxy."""
+        proxy.failure_count = 0
+        proxy.is_healthy = True
+    
+    def record_failure(self, proxy: ProxyConfig, error: Exception = None):
```

Ruff (0.14.1)

PEP 484 prohibits implicit `Optional`

Convert to `Optional[T]`

(RUF013)